File: apps/frontend/src/types/question.ts
export type ModuleKey = "integer-operations" | "algebra-expressions" | "algebra-formulas";
export type Difficulty = "easy" | "medium" | "hard";
export type InputType = "text" | "multiple_choice";

export interface ChoiceOption {
  key: string;
  label: string;
}

export interface Question {
  question_id: string;
  module: ModuleKey;
  difficulty: Difficulty;
  prompt: string;
  input_type: InputType;
  question_kind: string;
  placeholder?: string;
  choices?: ChoiceOption[];
  validation_token: string;
}

File: apps/backend/app/schemas/question.py
from typing import Literal

from pydantic import Field

from app.schemas.common import BaseSchema


class ChoiceOption(BaseSchema):
    key: str
    label: str


class QuestionResponse(BaseSchema):
    question_id: str
    module: Literal["integer-operations", "algebra-expressions", "algebra-formulas"]
    difficulty: Literal["easy", "medium", "hard"]
    prompt: str
    input_type: Literal["text", "multiple_choice"]
    question_kind: str
    placeholder: str | None = None
    choices: list[ChoiceOption] | None = None
    validation_token: str


class GenerateQuestionQuery(BaseSchema):
    module: Literal["integer-operations", "algebra-expressions", "algebra-formulas"]
    difficulty: Literal["easy", "medium", "hard"]
